app/src/main/java/com/github/libretube/ui/dialogs/DownloadPlaylistDialog.kt

```diff
@@ -0,0 +1,87 @@
+
+class DownloadPlaylistDialog : DialogFragment() {
+    private lateinit var playlistId: String
+    private lateinit var playlistName: String
+    private lateinit var playlistType: PlaylistType
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        playlistId = requireArguments().getString(IntentData.playlistId)!!
+        playlistName = requireArguments().getString(IntentData.playlistName)!!
+        playlistType = requireArguments().serializable(IntentData.playlistType)!!
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val binding = DialogDownloadPlaylistBinding.inflate(layoutInflater)
+
+        val possibleVideoQualities = resources.getStringArray(R.array.defres).toList().let {
+            // remove the automatic quality entry
+            it.subList(1, it.size)
+        }
+        val possibleAudioQualities = resources.getStringArray(R.array.audioQualityBitrates)
+        val availableLanguages = LocaleHelper.getAvailableLocales()
+
+        binding.videoSpinner.items = listOf(getString(R.string.no_video)) + possibleVideoQualities
+        binding.audioSpinner.items = listOf(getString(R.string.no_audio)) + possibleAudioQualities
+        binding.subtitleSpinner.items =
+            listOf(getString(R.string.no_subtitle)) + availableLanguages.map { it.name }
+        binding.audioLanguageSpinner.items =
+            listOf(getString(R.string.default_language)) + availableLanguages.map { it.name }
+
+        return MaterialAlertDialogBuilder(requireContext())
+            .setTitle(getString(R.string.download_playlist) + ": " + playlistName)
+            .setMessage(R.string.download_playlist_note)
+            .setView(binding.root)
+            .setPositiveButton(R.string.download) { _, _ ->
+                val maxVideoQuality = binding.videoSpinner.selectedItemPosition.let { videoPosition ->
+                    if (videoPosition >= 1) possibleVideoQualities[videoPosition - 1]
+                        .getWhileDigit() else null
+                }
+                val maxAudioQuality = binding.audioSpinner.selectedItemPosition.let { audioPosition ->
+                    if (audioPosition >= 1) possibleAudioQualities[audioPosition - 1]
+                        .getWhileDigit() else null
+                }
+                val captionLanguage = binding.subtitleSpinner.selectedItemPosition.let { captionPosition ->
+                    if (captionPosition >= 1) availableLanguages[captionPosition - 1].code else null
+                }
+                val audioLanguage = binding.audioLanguageSpinner.selectedItemPosition.let { audioLanguage ->
+                    if (audioLanguage >= 1) availableLanguages[audioLanguage - 1].code else null
+                }
+
+                if (maxVideoQuality == null && maxAudioQuality == null) {
+                    return@setPositiveButton
```

Comment:
In this instance, we leave the Dialog without informing the user about what happened. What do you think about a toast informing him that the download didn't start? 🤔

Reply:
:+1: